### DISCUSSION TITLE
i asked it to generate "-c 1980" "an article with a minimum of 1800 words" but sometimes it generate detailed article...

### DISCUSSION THREAD

**Discussion by hiqsociety** [ggml-org/llama.cpp#3201] [Q&A]
i asked it to generate "-c 1980" "an article with a minimum of 1800 words" but sometimes it generate detailed article, sometimes it generates single line.

how to make it consistently generate detailed encyclopedia styled article with a minimum of 1800 words?

any help is appreciated. thx in advance.

**Comment by KerfuffleV2** [comment]
It's really up to the model's training, its capabilities (don't expect so much from small models), how good your prompt is, etc. Also make sure you're setting `-n` high enough. `-c` sets the limit for the prompt + output, `-n` sets how many tokens to generate (after the prompt). The model may or may not generate that many tokens.

You can also try setting `--ignore-eos`. That will make it so the model can't signal that it's done and will keep generating tokens. You will get the tokens you asked for that way, but whether they actually are coherent is definitely not guaranteed.
